### PR TITLE
Replace \\e\sc\\ap\\\\ed\\\\\ regexp strings with `rx` forms.

### PR DESCRIPTION
**Pull request by MerelyAPseudonym**
There's still a little more I'd like to do to clean up, but I wanted to share the first iteration. (i.e., you almost certainly shouldn't merge this stuff yet)
If you guys like this overall change, I will of course squash these commits. They're broken up the way they are to make the pull request easier to review.

### CODE REVIEW

**Comment by bbatsov** [comment]
@MerelyAPseudonym Feel free to reopen this if you ever decide to update it.

**Comment by expez** [comment]
I just tried to poke at this area of `clojure-mode` and really wished this was merged.  Any time to revisit this @MerelyAPseudonym?

**Comment by MerelyAPseudonym** [comment]
@expez I actually have a day off today, so I'll see what I can do :wink:

File: clojure-mode.el [hunk 1]
```diff
@@ -77,228 +77,303 @@
...
+            (optional (submatch
+                       (one-or-more (syntax word)))))
        (1 font-lock-keyword-face)
        (2 font-lock-function-name-face nil t))
       ;; (fn name? args ...)
-      (,(concat "(\\(?:clojure.core/\\)?\\(fn\\)[ \t]+"
-                ;; Possibly type
-                "\\(?:#?^\\sw+[ \t]*\\)?"
-                ;; Possibly name
-                "\\(t\\sw+\\)?" )
```
Comment (expez) [inline clojure-mode.el]: That "t" looks like a typo.

File: clojure-mode.el [hunk 2]
```diff
@@ -77,228 +77,303 @@
 (defconst clojure-font-lock-keywords
   (eval-when-compile
     `( ;; Definitions.
-      (,(concat "(\\(?:clojure.core/\\)?\\("
-                (regexp-opt '("defn" "defn-" "def" "defonce"
-                              "defmulti" "defmethod" "defmacro"
-                              "defstruct" "deftype" "defprotocol"
-                              "defrecord" "deftest" "def\\[a-z\\]"))
```
Comment (expez) [inline clojure-mode.el]: This also seems to serve no purpose

File: clojure-mode.el [hunk 3]
```diff
@@ -77,228 +77,303 @@
...
-                ;; Function declarations.
-                "\\)\\>"
-                ;; Any whitespace
-                "[ \r\n\t]*"
-                ;; Possibly type or metadata
-                "\\(?:#?^\\(?:{[^}]*}\\|\\sw+\\)[ \r\n\t]*\\)*"
```
Comment (expez) [inline clojure-mode.el]: This regexp is also out of date.  A type hint typically looks like `^String` now. And not `#^String`.  Perhaps we should fix these later.

**Comment by MerelyAPseudonym** [comment]
btw, I'm keeping my progress in [a different branch](https://github.com/MerelyAPseudonym/clojure-mode/tree/font-lock-with-rx_v2) for the time being.

**Comment by expez** [comment]
@MerelyAPseudonym any vacation days coming up soon? :p

**Pull request closed without merging**